tsdb/chunks/chunks.go

```diff
@@ -190,40 +190,50 @@ func (w *Writer) cut() error {
 		return err
 	}
 
-	p, _, err := nextSequenceFile(w.dirFile.Name())
+	n, f, _, err := cutSegmentFile(w.dirFile, chunksFormatV1, true, w.segmentSize)
 	if err != nil {
 		return err
 	}
+	w.n = int64(n)
```

Comment:
Why we do this outside of cutSegment File?

Reply:
`cutSegmentFile` is shared by this struct and also the new `HeadReadWriter` that I introduced. It is a standalone function now and not a part of either of the struct. Hence need to do it outside.